### PR TITLE
fix(wallet-mobile): NetworkError instance

### PR DESCRIPTION
**Pull request by jorbuedo**
related: https://emurgo.atlassian.net/browse/YOMO-1356

This one was confusing. Not sure why we have `NetworkError` exported from `@yoroi/common`, but it's not the same one the wallet is throwing, so don't mix them up because then `instanceof` will fail.

### CODE REVIEW

**Comment by banklesss** [comment]
maybe it's worth to still use the common package error but catching it by instance and use the UI error messages?

**Comment by jorbuedo** [comment]
> maybe it's worth to still use the common package error but catching it by instance and use the UI error messages?

Well, sure, we can use the one in common everywhere instead. None of the uses I see use the message from the error itself.

File: apps/wallet-mobile/src/features/SetupWallet/useCases/CreateWallet/WalletDetailsScreen.tsx [hunk 1]
```diff
@@ -233,7 +233,7 @@ export const WalletDetailsScreen = () => {
   ])
 
   React.useEffect(() => {
-    if (showRestoreWalletInfoModal) showModalTipsPassword()
+    if (showRestoreWalletInfoModal === true) showModalTipsPassword()
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/CreateWallet/WalletDetailsScreen.tsx]:
```suggestion
    if (showRestoreWalletInfoModal) showModalTipsPassword()
```

**Comment by stackchain** [comment]
> > maybe it's worth to still use the common package error but catching it by instance and use the UI error messages?
> 
> Well, sure, we can use the one in common everywhere instead. None of the uses I see use the message from the error itself.

The network error inside wallet is legacy, there are 2 fetchers, the legacy fetch, and the fetchData, which is the new one. There will be few places to replace. Please update them all to use -> from types Api.Error.Network, from types, the errors inside common can be dropped.

File: apps/wallet-mobile/src/features/SetupWallet/useCases/CreateWallet/WalletDetailsScreen.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/features/SetupWallet/useCases/CreateWallet/WalletDetailsScreen.tsx]: Not sure why this was complaining for me, saying `showRestoreWalletInfoModal` was `any` and required a specific check, despite it being a boolean.

Comment (stackchain) [reply apps/wallet-mobile/src/features/SetupWallet/useCases/CreateWallet/WalletDetailsScreen.tsx]: I have weird messages on it too sometimes with VSCode + eslint, then I close (like windows) I close VSCode and open it again, it stops complaining, it looks this rule for boolean has come conflict with VSC or something is not working between them.

**Review by stackchain — APPROVED** [review]

**Pull request merged**